models/DataObject/ClassDefinition/Data/Time.php

```diff
@@ -113,16 +113,16 @@ public function checkValidity($data, $omitMandatoryCheck = false, $params = [])
             throw new Model\Element\ValidationException('Wrong time format given must be a 5 digit string (eg: 06:49) [ '.$this->getName().' ]');
         }
 
-        if (!$omitMandatoryCheck && strlen($data)) {
+        if (!$omitMandatoryCheck && is_string($data) && strlen($data)) {
```

Comment:
```suggestion
        if (!$omitMandatoryCheck && $data) {
```
Should be empty or a string at this position. Otherwise it should already thrown a exception at this position. So I think it is enough to check it without strlen()